src/spatial_reactionsystem.jl

```diff
@@ -0,0 +1,142 @@
+    spatial_reactions::Vector{SpatialReaction}
+    """The graph on which the lattice is defined."""
+    lattice::MetaGraph
+    """Dependent (state) variables representing amount of each species. Must not contain the
+    independent variable."""
+    
+    function LatticeReactionSystem(rs_base, spatial_reactions, lattice)
+        rs = make_spatial_reaction_system(rs_base, spatial_reactions, lattice)
+        return new(rs, rs_base, spatial_reactions, lattice)
+    end
+end
+
+
+# Function for creating a single ReactionSystem structure from a LatticeReactionSystem input.
+function make_spatial_reaction_system(rs::ReactionSystem,srs::Vector{SpatialReaction},lattice::MetaGraph)
+    # Extends an empty base network, with one copy of the input network for each node on the graph.
+    @named rs_base = ReactionSystem(rs.iv)
+    rs_combined = flatten(compose(rs_base, map(c -> make_subsystem(rs,c,lattice),vertices(lattice))))
+
+    # Loops through all connections, and spatial reactions, and add them (in both directions) for each connection.
+    spatial_reactions = Vector{Reaction}()
+    for sr in srs, e in edges(lattice)
+        # Could create a loop to generate all 8 expressions, but code is easier to read this way.
+        subs1_r1 = map(sub -> sym_to_var(sub,rs_combined,e.src,lattice), first.(sr.substrates[1]))
+        subs2_r1 = map(sub -> sym_to_var(sub,rs_combined,e.dst,lattice), first.(sr.substrates[2]))
+        subs1_r2 = map(sub -> sym_to_var(sub,rs_combined,e.dst,lattice), first.(sr.substrates[1]))
+        subs2_r2 = map(sub -> sym_to_var(sub,rs_combined,e.src,lattice), first.(sr.substrates[2]))
+        prods1_r1 = map(sub -> sym_to_var(sub,rs_combined,e.src,lattice), first.(sr.products[1]))
+        prods2_r1 = map(sub -> sym_to_var(sub,rs_combined,e.dst,lattice), first.(sr.products[2]))
+        prods1_r2 = map(sub -> sym_to_var(sub,rs_combined,e.dst,lattice), first.(sr.products[1]))
+        prods2_r2 = map(sub -> sym_to_var(sub,rs_combined,e.src,lattice), first.(sr.products[2]))
+
+        push!(spatial_reactions,Reaction(sr.rate,[subs1_r1;subs2_r1],[prods1_r1;prods2_r1],[last.(sr.substrates[1]);last.(sr.substrates[2])],[last.(sr.products[1]);last.(sr.products[2])]))
+        push!(spatial_reactions,Reaction(sr.rate,[subs1_r2;subs2_r2],[prods1_r2;prods2_r2],[last.(sr.substrates[1]);last.(sr.substrates[2])],[last.(sr.products[1]);last.(sr.products[2])]))
+    end
+
+    # Adds the spatial reactions to the initial network.
+    return extend(rs_combined,ReactionSystem(spatial_reactions,rs_combined.iv;name=rs_combined.name))
+end
+
+# Helper functions.
+get_name(comp::Int64,lattice::MetaGraph) = has_prop(lattice,comp,:name) ? get_prop(lattice,comp,:name) : Symbol("container___",comp)
+make_subsystem(rs::ReactionSystem,comp::Int64,lattice::MetaGraph) = @set! rs.name = get_name(comp,lattice)
+sym_to_var(sym,rs,comp,lattice) = rs.var_to_name[Symbol(get_name(comp,lattice),:₊,sym)]
```

Comment:
Use `get_var_to_name` instead of direct field access as it is part of the API. Saying `sys.field` is discouraged since `sys.field` is now supposed to represent accessing a symbolic variable/parameter, and not accessing an internal field.

Reply:
Good catch.